### PR TITLE
Generate output for ode models

### PR DESCRIPTION
**Pull request by richfitz**
Allows generation of output for ode models

### CODE REVIEW

File: R/parse_system.R [hunk 1]
```diff
@@ -62,6 +62,22 @@ parse_system_overall <- function(exprs, call) {
...
   }
 
+  if (any(is_output)) {
+    if (!is_continuous) {
+      odin_parse_error(
+        c("Can't use 'output()' in discrete time systems",
+          i = paste("You should be able to do what you need using 'update()'.",
+                    "If you are migrating from odin 1.x.x, you might find",
+                    'some advice in {.vignette "migrating"}')),
+        "E2999", src, call)
```
Comment (richfitz) [inline R/parse_system.R:72]:
```suggestion
        "E2019", src, call)
```

File: vignettes/errors.Rmd [hunk 2]
```diff
@@ -1015,3 +1015,7 @@ but these would not:-
 update(xy[]) <- 1
 xyz[1, 2] <- xyz[1, 2] + 1
 ```
+
+# `E2999`
```
Comment (richfitz) [inline vignettes/errors.Rmd:1019]:
```suggestion
# `E2019`
```

**Review by weshinsley — APPROVED** [review]
All looks good - couple of future tickets:-

1. We should add `output` to functions.Rmd as our kind of syntax definition.
2. In doing so - we could be particular about the types, and add some specific error messages maybe - taking this from the tests - 
```
dat <- odin_parse({
  initial(x[]) <- 0
  deriv(x[]) <- x * r[i]
  r <- parameter()
  n <- 3
  dim(x) <- n
  dim(r) <- n
  output(tot) <- sum(x)
})
```

If I change `tot` in the last line to stupid options (`r` or `n` for instance), the error messages could be more pointed.

**Comment by richfitz** [comment]
Added mrc-5948, mrc-5949 and mrc-5950 for these

**Pull request merged**